yarl/_url.py

```diff
@@ -142,7 +142,7 @@ class URL:
     _PATH_UNQUOTER = _Unquoter(unsafe="+")
     _QS_UNQUOTER = _Unquoter(qs=True)
 
-    def __new__(cls, val="", *, encoded=False, strict=None):
+    def __new__(cls, val="", *, encoded=False, strict=None, separator="&"):
```

Comment:
If we start providing this public API to the end-users, we must ensure that users calling it with `separator=";"` on old and new Pythons will get the same results regardless of whether the underlying function in stdlib supports this argument or not.

This means, that we need a compatibility shim in the `_compat` module normalizing the behavior across the board.